core/domain/exp_domain.py

```diff
@@ -1456,9 +1459,12 @@ def validate(self, strict: bool = False) -> None:
                         # we find a terminal state in an outcome, we break out
                         # of the for loop and raise a validation error.
                         all_outcomes = (
-                            curr_state.interaction.get_all_outcomes())  # type: ignore[no-untyped-call]
+                            curr_state.interaction.get_all_outcomes())
                         for outcome in all_outcomes:
                             dest_state = outcome.dest
+                            # Ruling out the possibility of None for mypy type
```

Comment:
You can't just do this, as explained before; you'll need an explanation about why we know for certain that dest is not None at this stage, otherwise we're setting ourselves up for a runtime error in production.

Reply:
In the same function ( validate ), we are checking that every answer's outcome should have a destination. 
https://github.com/oppia/oppia/blob/a65d963b3268c484a2fafa6778979a79c5effefe/core/domain/exp_domain.py#L1194

we are also checking for default_outcome, I mean if default_outcome exists for an interaction then it should have a destination:
https://github.com/oppia/oppia/blob/a65d963b3268c484a2fafa6778979a79c5effefe/core/domain/exp_domain.py#L1211

So, before reaching this code we are already validated that dest exists or not.

Reply:
Is your reasoning that (a) every outcome must have a destination from its moment of creation -- i.e. it's impossible for it not to have one, or (b) we have checked for it above?

Your code comment implies (a) but your argument above implies (b), so I'm not quite sure what the reasoning is and, therefore, how to respond to it.

Reply:
I was pointing to point (b) we have checked for it above?

Comment is updated!!